lib/vault/rails.rb

```diff
@@ -160,10 +161,13 @@ def memory_decrypt(path, key, ciphertext, client: , context: nil)
 
         return nil if ciphertext.nil?
 
+        raise Vault::Rails::InvalidCiphertext.new(ciphertext) if !ciphertext.start_with?(DEV_PREFIX)
```

Comment:
@malnick This is one thing that we may want to change.. this will throw an error for all old data without the prefix. Perhaps we want to handle it more gracefully and not throw an error if the prefix is missing?

Reply:
Looking at this, it appears that that method, `memory_decrypt`, should only be used when the vault server is running in `-dev` mode, which means that there shouldn't be any old data.

Reply:
Correction, that should only run when it is not actually talking to a vault server. It should never be run outside a testing/development context.

Reply:
Yes, that is correct. There may still be reasons to not break people's development environments. A note could be put in the changelog in case someone is affected and comes looking.